### DISCUSSION TITLE
Reflash Megadrive Repro from Aliexpress "Flashkit PCB#

### DISCUSSION THREAD

**Discussion by JuaninX** [sanni/cartreader#532] [General Chat]
HI!
I want to buy these Megadrive pcb from aliexpres, can they be rewritten?

![H5dc931bc065546dfb24a6e8a8b270b93f](https://user-images.githubusercontent.com/109401405/192102584-8c33b80b-75cb-43d3-a9a6-0d9c021475b7.png)
![Tiny-Toon-Adventures-ACME-All-Stars-EUR-etiqueta-Flashkit-tarjeta-MD-para-Sega-Genesis-Megadrive-consola](https://user-images.githubusercontent.com/109401405/192102588-507a6485-af94-4b44-b6a7-27d34570cb47.png)

they sell this cartwritter

![KY-Technology-FlashKit-programmer-MD-FlashKit-Card-For-Sega-Megadrive-Genesis-Video-Game-Console-Game-Cartridge](https://user-images.githubusercontent.com/109401405/192102602-fffcef5e-029d-42f1-b677-1425fc21dc7a.png)

**Comment by sanni** [comment]
I have only added support for 29F1610 flashroms in the Mega Drive code. You would need to write your own functions to flash the M29W640T on that repro.

**Reply by JuaninX** [reply]
Thanks!! I will try.

**Reply by sanni** [reply]
Most of the commands are actually the same as with the 29F1610. But the datasheet doesn't seem to show the Flash ID command only the CFI command. So that would be the first thing to try, see if you get the proper flash ID. Erasing should then work with the current code but the writing function would need to be modified as the M29 doesn't have a full write buffer but writes 4 bytes in sequence. Probably a bit of trial and error until it will work but I think its doable with a reasonable amount of effort.

**Comment by JuaninX** [comment]
I understand, when I get the pcb, I'll do tests, if I succeed, I'll put the code here, thank you very much.

**Reply by Aergan** [reply]
I don't suppose you were successful with this perchance? Interested in the same solution

**Comment by Metro-City** [comment]
Hi, do you know what this error means: Verify error at 288
